### DISCUSSION TITLE
JavaScript Logic Challenges

### DISCUSSION THREAD

**Discussion by thanhvu85** [community/community#140236] [Programming Help]
How can you check if a variable is NaN (Not-a-Number) in JavaScript without using the isNaN() function?

**Comment by Kartikey2011yadav** [comment]
Since NaN is the only JavaScript value that is treated as unequal to itself, you can also test if a value is NaN by checking it for equality to itself:

```
var a = NaN;
a !== a; // true 
```

**Comment by abishek1740P** [comment]
You can check if a variable is NaN in JavaScript without using the isNaN() function by using the fact that NaN is the only value in JavaScript that is not equal to itself

```javascript
function isReallyNaN(value) {
    return value !== value;
}

console.log(isReallyNaN(NaN));       // true
console.log(isReallyNaN(123));       // false
console.log(isReallyNaN("string"));  // false

```

**Comment by queenofcorgis** [comment]
Hey there!  👋 

Thanks for posting in the GitHub Community, @maiidaothanhvu ! We're happy you're here. You are more likely to get a useful response if you are posting your question in the applicable category. The Accessibility category is a place for our community to discuss and provide feedback on the digital accessibility of GitHub products. Digital accessibility means that GitHub tools, and technologies, are designed and developed so that people with disabilities can use them.

We’ve moved your post to our Programming Help 🧑‍💻 category, which is more appropriate for this type of discussion.

Please review our [guidelines](https://github.com/orgs/community/discussions/51384) about the Programming Help category for more information.

**Comment by RexActor** [comment]
you can use `Object.Is` function  as well
`let value =1;
console.log(Object.is(value,NaN)); `

**Comment by luoyerr** [comment]
使用`String`进行比较
```
function isReallyNaN(value) {
    return String(value) === 'NaN';
} 
console.log(isReallyNaN(NaN));
console.log(isReallyNaN(0 / 0));
console.log(isReallyNaN('abc'));
console.log(isReallyNaN(String));
```
